tests/unittests/algo/test_dehb.py

```diff
@@ -0,0 +1,126 @@
+"""Perform integration tests for `orion.algo.dehb`."""
+
+import itertools
+
+import numpy
+import pytest
+
+from orion.algo.dehb.dehb import IMPORT_ERROR, UnsupportedConfiguration
+from orion.core.utils import backward, format_trials
+from orion.testing.algo import BaseAlgoTests
+
+if IMPORT_ERROR:
+    pytest.skip("skipping DEHB tests", allow_module_level=True)
+
+
+class TestDEHB(BaseAlgoTests):
+    """Test suite for algorithm DEHB"""
+
+    algo_name = "dehb"
+    config = {
+        "seed": 1234,
+        # Because this is so random
+        # Add other arguments for your algorithm to pass test_configuration
+        "mutation_factor": 0.65,
+        "crossover_prob": 0.45,
+        "mutation_strategy": "rand2dir",
+        "crossover_strategy": "exp",
+        "boundary_fix_type": "clip",
+        "min_clip": None,
+        "max_clip": None,
+        "max_age": 12,
+    }
+    space = {"x": "uniform(0, 1)", "y": "uniform(0, 1)", "f": "fidelity(1, 10, base=2)"}
+
+    def test_config_mut_strategy_isnot_valid(self):
+        with pytest.raises(UnsupportedConfiguration):
+            self.create_algo(config=dict(mutation_strategy="123"))
+
+    def test_config_cross_strategy_isnot_valid(self):
+        with pytest.raises(UnsupportedConfiguration):
+            self.create_algo(config=dict(crossover_strategy="123"))
+
+    def test_config_fix_mode_isnot_valid(self):
+        with pytest.raises(UnsupportedConfiguration):
+            self.create_algo(config=dict(boundary_fix_type="123"))
+
+    def test_missing_fidelity(self):
+        with pytest.raises(RuntimeError):
+            space = self.create_space(dict(x="uniform(0, 1)"))
+            self.create_algo(space=space)
+
+    def test_suggest_n(self, mocker, num, attr):
+        algo = self.create_algo()
+        spy = self.spy_phase(mocker, num, algo, attr)
+        trials = algo.suggest(3)
+        assert len(trials) == 3
+
+    @pytest.mark.xfail
+    def test_is_done_cardinality(self):
+        """Fails because of https://github.com/Epistimio/orion/issues/598"""
+        space = self.update_space(
```

Comment:
Question: In this test, why is it doing `space = self.update_space` if it's going to overwrite all the dimensions of the space anyway?

Reply:
Oh, first `space` is a configuration, a dictionary. I improved the var names in aeb1d570.

Reply:
No I understand, what I'm asking is, why is the `space_dict` created using `self.update_space`, instead of just a dict? (Since it's overwriting all the entries in `type(self).space` anyway)?